backend/src/middleware/validation.ts

```diff
@@ -30,11 +41,51 @@ const validateRequest: RequestHandler = (req, res, next) => {
   throw createHttpError(400, errorString);
 };
 
+const validateUser: RequestHandler = asyncHandler(async (req, res, next) => {
+  const authHeader = req.headers.authorization;
+  let token;
+
+  if (authHeader && authHeader.split(" ")[0] === "Bearer") {
+    token = authHeader.split(" ")[1];
+  } else {
+    throw createHttpError(401, "Authorization Token Required");
+  }
+
+  try {
+    const decodedToken = await firebaseAdmin.firebaseAuth.verifyIdToken(token);
+    const email = decodedToken.email;
+
+    const user = await UserModel.findOne({ email });
+    /* eslint-disable-next-line @typescript-eslint/no-unsafe-member-access */
+    req.body.currentUser = user;
+  } catch (error) {
+    throw createHttpError(404, "Error finding user");
+  }
+
+  next();
+});
+
+const validateHousingLocator: RequestHandler = (req, res, next) => {
+  validateUser(req, res, () => {
+    /* eslint-disable-next-line @typescript-eslint/no-unsafe-member-access */
+    if (req.body.currentUser.isHousingLocator) {
```

Comment:
I think we should inject `currentUser` into `req` instead of `req.body` so we don't accidentally clash with what might be in the `req.body` of the actual request.

```suggestion
    req.currentUser = user;
  } catch (error) {
    throw createHttpError(404, "Error finding user");
  }

  next();
});

const validateHousingLocator: RequestHandler = (req, res, next) => {
  validateUser(req, res, () => {
    /* eslint-disable-next-line @typescript-eslint/no-unsafe-member-access */
    if (req.currentUser.isHousingLocator) {
```